aviary/utils/doctape.py

```diff
@@ -435,3 +444,156 @@ def glue_keys(dict_of_dicts: dict, display=True) -> list:
+
+    Parameters
+    ----------
+    cmd: str
+        Aviary command
+    curr_glued: list
+        the parameters that have been glued
+    glue_default: boolean
+        flag whether the default values should be glued.
+    """
+    parser = argparse.ArgumentParser()
+    _command_map[cmd][0](parser)
+    actions = [*parser._get_optional_actions(), *parser._get_positional_actions()]
+    for action in actions:
+        opt_list = action.option_strings
+        for opt in opt_list:
+            if opt not in curr_glued:
+                glue_variable(opt, md_code=md_code)
+                curr_glued.append(opt)
+        if action.dest not in curr_glued:
+            glue_variable(action.dest, md_code=md_code)
+            curr_glued.append(action.dest)
+        if glue_default:
+            if str(action.default) not in curr_glued:
+                glue_variable(str(action.default), md_code=True)
+                curr_glued.append(str(action.default))
+        if glue_choices:
+            if action.choices is not None:
+                for choice in action.choices:
+                    if str(choice) not in curr_glued:
+                        glue_variable(str(choice), md_code=True)
+                        curr_glued.append(str(choice))
+
+
+def glue_class_functions(obj, curr_glued=None, pre_fix=None, md_code=True):
+    """
+    Glue all class functions
+
+    Parameters
+    ----------
+    obj: class
+        class object
+    curr_glued: list
+        the parameters that have been glued
+    """
+    methods = inspect.getmembers(obj, predicate=inspect.isfunction)
+    for func_name, func in methods:
+        if pre_fix is not None:
+            if pre_fix + '.' + func_name + '()' not in curr_glued:
+                glue_variable(pre_fix + '.' + func_name + '()', md_code=md_code)
+                curr_glued.append(pre_fix + '.' + func_name + '()')
+        if func_name + '()' not in curr_glued:
+            glue_variable(func_name + '()', md_code=md_code)
+            curr_glued.append(func_name + '()')
+
+
+def glue_function_arguments(func, curr_glued=None, glue_default=False, md_code=False):
+    """
+    Glue all function arguments and default values for a given function
+
+    Parameters
+    ----------
+    func: function
+        function
+    curr_glued: list
+        the parameters that have been glued
+    """
+    if curr_glued is None:
+        curr_glued = []
+    sig = inspect.signature(func)
+    for param_name, param in sig.parameters.items():
+        if param_name not in curr_glued and param_name != 'self':
+            glue_variable(param_name, md_code=md_code)
+            curr_glued.append(param_name)
+            if glue_default:
+                if str(param.default) is not param.empty:
+                    if str(param.default) not in curr_glued:
+                        glue_variable(str(param.default), md_code=md_code)
+                        curr_glued.append(str(param.default))
```

Comment:
```suggestion
            if glue_default and str(param.default) is not param.empty:
                param_default = param_name + '_default'
                if param_default not in curr_glued:
                    glue_variable(param_default, str(param.default), md_code=md_code)
                    curr_glued.append(param_default)
```
same comment as before about tracking where the default value came from

Reply:
I am not sure about this change. A `param` could be a string, an int, or boolean etc. I don't think it will work. But I did combine the first two "if"s to one "if".

Reply:
the value `param.default` can be any of those things, but `param.name` should always be a string.

Reply:
for example if you were writing about the `add_meta_data` function, and you wanted to mention that the default value for the `multivalue` argument was `False`, in the markdown cell, you would write:
```
The default value is {glue:md}`multivalue_default`
```
instead of:
```
The default value is {glue:md}`False`
```
otherwise, if the default is update to `True` in the code, the docs won't actually update to reflect that

Reply:
Done.